src/DotNetty.Transport/Channels/Sockets/TcpServerSocketChannel.cs

```diff
@@ -179,24 +203,29 @@ public override void FinishRead(SocketChannelAsyncOperation operation)
                     {
                         closed = true;
                     }
+                    catch (SocketException ex) when (ex.SocketErrorCode == SocketError.OperationAborted)
+                    {
+                        closed = true;
+                    }
+                    catch (SocketException ex)
+                    {
+                        exception = ex;
+                    }
                     catch (Exception ex)
                     {
-                        var asSocketException = ex as SocketException;
-                        if (asSocketException == null || asSocketException.SocketErrorCode != SocketError.WouldBlock)
+                        Logger.Warn("Failed to create a new channel from an accepted socket.", ex);
+                        exception = ex;
+
+                        if (connectedSocket != null)
```

Comment:
let's always dispose connectedSocket for socket exceptions if its not null.

basically, we can just remove the catch (SocketException ex) block